app/code/core/Mage/Core/Model/Config.php

```diff
@@ -502,17 +502,18 @@ public function getCache()
      */
     public function getCacheSaveLock($waitTime = null, $ignoreFailure = false)
     {
-        if (! Mage::app()->useCache('config')) {
+        if (!Mage::app()->useCache('config')) {
             return;
         }
-        $waitTime = $waitTime ?: (PHP_SAPI === 'cli' ? 60 : 3);
+        $waitTime = $waitTime ?: (getenv('MAGE_CONFIG_CACHE_LOCK_WAIT') ?: (PHP_SAPI === 'cli' ? 60 : 3));
```

Comment:
are we leaving 3 seconds as a default?

Reply:
Open to changing, what's your proposal and justification?

Reply:
After re-reading your previous comment:
> actually I think it's good to have a slightly bigger timeout because it forbids other requests (when timeout expires) from restarting the cache generation, that's what crashes servers, much more than requests piling up, at least in my experience

When the timeout occurs now, the timed out processes do not start saving the cache but rather they will exit with a 503 error. So, there is no risk of stampede but the timed out requests will obviously fail instead of potentially exhausting some other resource like PHP processes, RAM, web server connections, etc. In an ideal situation the systems are tuned so that the web server will handle these limits before they become a problem but a perfectly tuned server is probably pretty rare so I think it is safer to just consider 3 seconds to be a reasonable time to wait before failing. Adding the env var allows someone to adjust to their liking but I'm happy to concede that there is a better default value than 3 seconds if you make a good case. :)

Reply:
Ideally 3 seconds is enough, but "real world poorly configured stores" I think it's borderline, I'd much rather prefer 10 seconds but actually it was already merged so...